lib/Sema/CSSimplify.cpp

```diff
@@ -960,9 +960,37 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
 
   for (unsigned paramIdx = 0, numParams = parameterBindings.size();
        paramIdx != numParams; ++paramIdx){
-    // Skip unfulfilled parameters. There's nothing to do for them.
-    if (parameterBindings[paramIdx].empty())
+    // If a parameter is unfulfilled, validate the default argument if it is a
+    // magic literal expression.
+    if (parameterBindings[paramIdx].empty()) {
+      if (!callee)
+        continue;
+
+      if (!isa<AbstractFunctionDecl>(callee) ||
+          !(isa<EnumElementDecl>(callee)) || !isa<SubscriptDecl>(callee))
```

Comment:
You can use `!callee->hasParameterList()` here :)

Reply:
Ah, definitely simpler. Thanks!